R/check_params.R

```diff
@@ -127,6 +130,32 @@ check_params = function(params_sleep = c(), params_metrics = c(),
     check_class("general", params = params_general, parnames = numeric_params, parclass = "numeric")
     check_class("general", params = params_general, parnames = boolean_params, parclass = "boolean")
     check_class("general", params = params_general, parnames = character_params, parclass = "character")
+
+    ws3 = params_general[["windowsizes"]][1]; ws2 = params_general[["windowsizes"]][2]; ws = params_general[["windowsizes"]][3]
+    if (ws2/60 != round(ws2/60)) {
+      ws2 = as.numeric(60 * ceiling(ws2/60))
+      warning(paste0("The long windowsize needs to be a multitude of 1 minute periods.\n",
+                     "Long windowsize has now been automatically adjusted to ",
+                     ws2, " seconds in order to meet this criteria."), call. = FALSE)
+    }
+    if (ws2/ws3 != round(ws2/ws3)) {
+      def = c(1,5,10,15,20,30,60)
+      def2 = abs(def - ws3)
+      ws3 = as.numeric(def[which(def2 == min(def2))])
+      warning(paste0("The long windowsize needs to be a multitude of short windowsize.\n",
+                     "The short windowsize has now been automatically adjusted to ",
+                     ws3, " seconds in order to meet this criteria.\n"), call. = FALSE)
+    }
+
+    ws4 = 10 #epoch for recalibration, don't change
+    if (ws/ws4 != round(ws/ws4)) {
+      ws = as.numeric(ws4 * ceiling(ws/ws4))
```

Comment:
I did see in your comment in the code that `ws4` was a constant, not to be changed. But the way g.calibrate code is written, `ws` has to be a multiple of `ws4`. Otherwise the code crashes doing `dim(Gx)`, `dim(Gy)`, `dm(Gz)` assignments [here](https://github.com/wadpac/GGIR/blob/04f6c53593da3cf2e441924e3f083a92375d6e03/R/g.calibrate.R#L171-L173) 

I was doing some testing, and for some reason I entered ws=1805 in one of my calls to GGIR, and GGIR crashed. I think it's unlikely people would want to use an odd windowsize like this, but I still think it's better if code handles most inputs gracefully and doesn't crash.

So I added the logic to check that `ws` is indeed a multiple of `ws4`. But I wanted to do so in a central location, not inside `g.calibrate`. Because if I adjust `ws` in `g.calibrate, then `g.getmeta` and `g.calibrate` will use different values for `ws`, and that doesn't seem proper. So I decided to validate ws inside `check_params()` because it just seems like a logical place to validate parameter values. But then to validate `ws` there, I had to make `check_params()` aware of the `w4` constant.

So `ws4` is in `check_params` not because its value might change, but only because we need this constant to validate `ws`.

Please let me know this still doesn't seem right.

Reply:
Ok, I now understand what you mean. In that case I propose disentangling `windowsizes` in `g.calibrate` and `g.getmeta` by hard-coding `ws=3600` and `ws4 =10` in `g.calibrate`. There is no reason to align them with the `windowsizes` values in `g.getmeta`, because there is no evidence that different `windowsizes` lead to better autocalibration results. We could justify this by referring to descriptive article I wrote (2014) where those values were used.

While doing that I propose factoring out the need for `ws2` in g.calibrate, because in `g.downsample` it is used to calculate `var3` but `var3` is not used in `g.calibrate` by which `g.downsample` can skip its calculation if `ws2` is NULL, where we need set `ws2 = NULL` as default for `g.downsample`. In that way `g.calibrate` does not have to refer to `ws2` any longer.

Further, it may be good to rename `ws4` to `calibrationEpochSize`, and `ws` to `LongWindowSize` or something similar to make code more readable.

The above hopefully helps to simplify g.calibrate. I am happy to make a PR into your PR for these suggestion if you like.

Reply:
I see, this sounds good.
If you would make the PR, I would really appreciate it! This way I don't misunderstand some part of your suggested improvement.